task3_2/js/src/function.js

```diff
@@ -0,0 +1,131 @@
+//isArray function
+function isArrayPrototype(param) {
+  return Object.prototype.toString.call(param) === '[object Array]';
+}
+
+function isArrayInsanceof(param) {
+  return param instanceof Array;
+}
+
+function isArrayConstructor(param) {
+  return param && param.constructor === Array;
+}
+
+//range function
+function range(left, right, step = 1) {
```

Comment:
next cases should work:
range(10) [0..9]
range(1, 10) [1..9]
range(1, 10, 3) [1, 4, 7]
range(10, null, 3) [0, 3, 6, 9]
range(10, null) the same with range(10)
range(-2, -5) []
range(-5, -2) [-5, -4, -3]
range(-5, -2, 2) [-5, -3]
range(-5, -2, -2) []
range(-5, null, 2) []
range(-5, null, -2) [0, -2, -4]
range(-10, -20, -5) [-10, -15]
range(-20, -10, -5) []
some of them don't work with your implementation

suggestion for you: you should define length of result array for short and cool implementation